### ISSUE TITLE
Generation of random unitary matrix does not draw matrices from Haar measure

### ISSUE THREAD

**Issue by pgawron** [QISKit/qiskit-sdk-py#519]
https://github.com/QISKit/qiskit-sdk-py/blob/5c1c68a5aa3dcccdf5c10f9eb307383ebb40826b/qiskit/tools/qi/qi.py#L346

The method of generation random unitary matrices is faulty. It will not generate random matrices drawn from the Haar measure.

Please refer to  "How to generate random matrices from the classical compact groups", Francesco Mezzadri, arXiv:math-ph/0609050 p.11.

**Comment by nonhermitian** [comment]
I would not say it is faulty.  It just doesn't draw from the Haar measure.

**Comment by jaygambetta** [comment]
I agree with @pgawron we should update this to be from the Haar measure. After we do this i agree @nonhermitian we should also make the documentation clear and say a random martix from the Haar measure.

**Comment by delapuente** [comment]
If this is addressable by improving the documentation, let's do it as soon as possible. Is this the case?

**Comment by jaygambetta** [comment]
No it needs to write a new algorithm. Its simple but I need to find some time

**Comment by ikkoham** [comment]
How about using scipy.stats.unitary_group?

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.unitary_group.html

**Comment by jaygambetta** [comment]
This gives you the unitary but in general it’s hard to decompose it into gates.  

The more I think about this let’s just change the name of the function.

**Comment by jaygambetta** [comment]
@ikkoham i agree. Can you just submit a Pr and then we can close this issue.  I would put a test with it as well.

**Comment by ikkoham** [comment]
Thanks. OK, I will submit a PR.
